client/lib/importer/actions.js

```diff
@@ -56,7 +57,7 @@ export function cancelImport( siteId, importerId ) {
 		.catch( apiFailure );
 }
 
-export function failUpload( importerId, error ) {
+export function failUpload( importerId, { message: error } ) {
```

Comment:
Mmm... I don't feel strongly. I think it's less clear to do your destructuring inside a function parameter, so to my taste it would be a little clearer the other way. But maybe I just need to see this more often to feel comfortable with it.